Tasks/NodeTool/nodetool.ts

```diff
@@ -0,0 +1,154 @@
+//import toolLib = require('vsts-task-tool-lib/tool');
+//import taskLib = require('vsts-task-lib/task');
+import * as toolLib from 'vsts-task-tool-lib/tool';
+import * as taskLib from 'vsts-task-lib/task';
+import * as restm from 'typed-rest-client/RestClient';
+import * as os from 'os';
+import * as path from 'path';
+
+let osPlat: string = os.platform();
+let osArch: string = os.arch();
+
+async function run() {
+    try {
+        let versionSpec = taskLib.getInput('versionSpec', true);
+        let onlyLTS: boolean = taskLib.getBoolInput('onlyLTS', false);
+
+        await getNode(versionSpec, onlyLTS);
+    }
+    catch (error) {
+        console.error('ERR:' + error.message);
+    }
+}
+
+//
+// Node versions interface
+// see https://nodejs.org/dist/index.json
+//
+interface INodeVersion {
+    version: string,
+    lts: any,
+    files: string[]
+}
+
+//
+// Basic pattern:
+//     version = find and evaluate local versions
+//          use latest match 
+//     if version not found locally
+//          // let's query
+//          if versionSpec is explicit version
+//               versionToGet = versionSpec
+//          else
+//               versionToGet = query and evaluate internet tool provider
+//          
+//          download versionToGet
+//          Extract or move to cache download
+//
+//      find tool path by version
+//      prepend $PATH with toolpath
+//
+async function getNode(versionSpec: string, onlyLTS: boolean) {
+    //
+    // Let's try and resolve the versions spec locally first
+    //
+    let localVersions: string[] = toolLib.findLocalToolVersions('node');
+    let version: string = toolLib.evaluateVersions(localVersions, versionSpec);
+
+    if (version) {
+        console.log('Resolved from tool cache: %s', version);
```

Comment:
todo loc. Or just print the file path with no message, then no need for a loc string.